lib/evaluate/check-call.cc

```diff
@@ -146,42 +149,53 @@ static void RearrangeArguments(const characteristics::Procedure &proc,
       messages.Say(*x.keyword,
           "Argument keyword '%s=' is not recognized for this procedure reference"_err_en_US,
           *x.keyword);
+      return false;
     }
   }
+  return true;
+}
+
+bool CheckExplicitInterface(const characteristics::Procedure &proc,
+    ActualArguments &actuals, FoldingContext &context) {
+  if (!RearrangeArguments(proc, actuals, context.messages())) {
+    return false;
+  }
+  int index{0};
+  for (auto &actual : actuals) {
+    const auto &dummy{proc.dummyArguments[index++]};
+    if (actual.has_value()) {
+      CheckExplicitInterfaceArg(*actual, dummy, context);
```

Comment:
`CheckExplicitInterfaceArg` should probably now return a Boolean too, so that you can `return false;` when it encounters an error.

Reply:
OK